### ISSUE TITLE
`spectrum.flux /= x` raises error, but still changes value

### ISSUE THREAD

**Issue by hamogu** [astropy/specutils#415]
Te code below follows the docs for continuum fitting (https://specutils.readthedocs.io/en/latest/fitting.html#continuum-fitting ) except that I also try to normalize a spectrum by the continuum:
```
import numpy as np
import matplotlib.pyplot as plt

from astropy.modeling import models
from astropy import units as u

from specutils.spectra import Spectrum1D, SpectralRegion
from specutils.fitting import fit_generic_continuum

np.random.seed(0)
x = np.linspace(0., 10., 200)
y = 3 * np.exp(-0.5 * (x - 6.3)**2 / 0.1**2)
y += np.random.normal(0., 0.2, x.shape)

y_continuum = 3.2 * np.exp(-0.5 * (x - 5.6)**2 / 4.8**2)
y += y_continuum

spectrum = Spectrum1D(flux=y*u.Jy, spectral_axis=x*u.um)

g1_fit = fit_generic_continuum(spectrum)

y_continuum_fitted = g1_fit(x*u.um)

plt.plot(x, y)
plt.plot(x, y_continuum_fitted)
plt.title('Continuum Fitting')
plt.grid(True)
```

So far, so good. 
Now I can print the flux values in the spectrum:
```
spectrum.flux
[1.9730844, 1.7201267, 1.8557246, …, 2.3072277, 2.5550603, 2.3695634]Jy
```
I now try to normalize the spectrum by deviding the flux by the fitted continuum values. I nthe docs, I can't find a description how to do that, so just do the naive thing:
```
spectrum.flux /= y_continuum_fitted
```
However, that raises an error:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-90-c95d71413366> in <module>()
----> 1 spectrum.flux /= y_continuum_fitted

AttributeError: can't set attribute
```

So, question 1: How am I supposed to do a continuum normalization? I see that there are a few other issues looking into that e.g. #231. 

However, even more surprising, despite raising a `ValueError` it apparently worked ?!?
```
spectrum.flux
[1.1513992, 0.99465468, 1.0632936, …, 1.2593426, 1.4306224, 1.3625425]Jy
```
How can a statement raise an error and still be successful?

**Comment by nmearl** [comment]
Arithmetic operations are performed on the `Spectrum1D` object itself, not on the (immutable) attributes like `flux`, `wavelength`, `velocity`, `spectral_axis`, etc.

```python
spectrum /= y_continuum_fitted
```

> However, even more surprising, despite raising a ValueError it apparently worked ?!?

This is an upstream issue -- I've been chatting with `NDData` devs here at the astropy conference. Hopefully fixed soon!

**Comment by mjrutkowski** [comment]
If we wanted to normalize by a single constant value, why does the spec.spectral_axis revert to its initial definition on the call to Spectrum1D()?  If I apply "shift to redshift" to shift the spectrum1d to z=0, and then divide by a constant (the flux at a some wavelength) spectral_axis reverts to the original, not z =0, spectral_axis. WHY?!

**Comment by rosteen** [comment]
> If we wanted to normalize by a single constant value, why does the spec.spectral_axis revert to its initial definition on the call to Spectrum1D()? If I apply "shift to redshift" to shift the spectrum1d to z=0, and then divide by a constant (the flux at a some wavelength) spectral_axis reverts to the original, not z =0, spectral_axis. WHY?!

FWIW there was an open issue for this (#1131), and if there hadn't been, it's probably better to open a new issue than comment on a very old closed issue. Regardless, thanks for the nudge - I addressed this in #1158 which I hope to have merged soon to include in the next release (which I was hoping to get out this week).
